src/engraving/libmscore/scorefont.cpp

```diff
@@ -0,0 +1,769 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "scorefont.h"
+
+#include "draw/painter.h"
+#include "log.h"
+#include "mscore.h"
+#include "sym.h"
+
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonParseError>
```

Comment:
Better to move these headers above our headers

Reply:
I will do that. But what is our "official" policy with headers? Is it like the following?
1. Corresponding header for cpp file
2. Qt / STL / other "foreign" headers
3. our own headers

Or should 1 and 2 be the other way round?

And I assume that it is preferred to keep them sorted alphabetically when there are many includes.

Reply:
I like this order
1. Corresponding header for cpp file
2. STL
3. Qt
4. 3rd party
5. libmscore
6. MU4 classes
   a. global
   c. other modules
   d. internal module

Please only not alphabetically

Reply:
```
Corresponding header for cpp file
Qt / STL / other "foreign" headers
our own headers
```
Right!
You can make an alphabetical order within the group :)

On the good, we need to swap 1 and 2, but historically it turned out as you wrote.
It is convenient to group STL, Qt, ours by modules ...
This order is also related to the fact that various definitions can be defined, which we do not even know about, but can affect the behavior of "foreign" headers. Better the way "foreign" definitions affect us, we will find out and fix it faster. For example https://github.com/musescore/MuseScore/blob/master/src/framework/ui/view/iconcodes.h#L42

Reply:
Thanks! I'll keep this in mind in the future :)
Maybe we should mention this in the Wiki (https://github.com/musescore/MuseScore/wiki/CodeGuidelines#header-and-implementation).